### PR TITLE
 [routing] Fix inadequate routing manoeuvres near the mwm border

### PR DESCRIPTION
**Pull request by dobriy-eeh**
Хакфикс странных маневров возле границ mwm.
Полноценное решение: избавиться от OSRM, а пока так.

PTAL @bykoianko

### CODE REVIEW

File: routing/car_router.cpp [hunk 1]
```diff
@@ -466,7 +466,12 @@ CarRouter::ResultCode CarRouter::CalculateRoute(m2::PointD const & startPoint,
     }
     INTERRUPT_WHEN_CANCELLED(delegate);
 
-    if (crossCode == NoError && crossDistanceM < route.GetTotalDistanceMeters())
+    // Cross mwm and single mwm routes are build by different algorithm.
+    // It is not correct to compare distances from different algorithms.
+    // Add some penalty to cross mwm route to avoid inadequate manoeuvres near the mwm borders.
+    // TODO: remove this hack with OSRM routing.
+    double constexpr crossMwmPenalty = 0.8;
```
Comment (mpimenov) [inline routing/car_router.cpp]: s/crossMwmPenalty/kCrossMwmPenalty

File: routing/car_router.cpp [hunk 2]
```diff
@@ -466,7 +466,12 @@ CarRouter::ResultCode CarRouter::CalculateRoute(m2::PointD const & startPoint,
     }
     INTERRUPT_WHEN_CANCELLED(delegate);
 
-    if (crossCode == NoError && crossDistanceM < route.GetTotalDistanceMeters())
+    // Cross mwm and single mwm routes are build by different algorithm.
```
Comment (mpimenov) [inline routing/car_router.cpp]: s/build/built/

Comment (mpimenov) [inline routing/car_router.cpp]: s/algorithm/algorithms/

File: routing/car_router.cpp [hunk 3]
```diff
@@ -466,7 +466,12 @@ CarRouter::ResultCode CarRouter::CalculateRoute(m2::PointD const & startPoint,
...
+    // It is not correct to compare distances from different algorithms.
+    // Add some penalty to cross mwm route to avoid inadequate manoeuvres near the mwm borders.
+    // TODO: remove this hack with OSRM routing.
```
Comment (mpimenov) [inline routing/car_router.cpp]: s/with/together with/

**Comment by mpimenov** [comment]
Add the [routing] tag to commit and PR, please.

**Comment by mpimenov** [comment]
LGTM after fixes

**Comment by bykoianko** [comment]
LGTM

File: routing/car_router.cpp (hunk 2 above)
Comment (dobriy-eeh) [inline routing/car_router.cpp]: fixed fixed

File: routing/car_router.cpp (hunk 3 above)
Comment (dobriy-eeh) [inline routing/car_router.cpp]: fixed

**Comment by dobriy-eeh** [comment]
All fixed.
@bykoianko @mpimenov PTAL & merge

**Pull request merged**